### ISSUE TITLE
Podman compose setup fails on Podman Desktop for windows

### ISSUE THREAD

**Issue by sockejr** [containers/podman-desktop#5776]
### Bug description

I tried to execute compose setup on the GUI via Settings/Resources compose. After asking for admin credentials Docker Desktop shows the screen: Failed Installing Compose and a Button Try again.
Any hints for troubleshooting are appreciated.

### Operating system

Windows10

### Installation Method

Installer from website/GitHub releases

### Version

Podman Desktop 1.7.0 / 1.7.1 , Podman 4.8.0/ 4.9.2

### Steps to reproduce

as above

### Relevant log output

```shell
where to find log output ?
```

**Comment by franoisD** [comment]
same issue..

**Comment by ZPOMI** [comment]
same issue on Podman Desktop for Windows(v1.7.1)

**Comment by ZPOMI** [comment]
same issue on Podman Desktop for Windows(v1.7.1)

**Comment by sockejr** [comment]
Any hints on how to troubleshoot are welcome

**Comment by odockal** [comment]
@sockejr @ZPOMI @franoisD thanks for reporting.  
If you open developer tools in Podman Desktop (CTRL + SHIFT + I), open console, there could be an error when executing the workflow, can you please past it here. Or alternatively, open troubleshooting page and check Logs (PD status bar light bulb icon -> Logs).

**Comment by jeffmaury** [comment]
Cannot reproduce on my Win11Pro machine

**Comment by ZPOMI** [comment]
> @sockejr @ZPOMI @franoisD thanks for reporting. If you open developer tools in Podman Desktop (CTRL + SHIFT + I), open console, there could be an error when executing the workflow, can you please past it here. Or alternatively, open troubleshooting page and check Logs (PD status bar light bulb icon -> Logs).

Thank you very much for your work. I'm sorry for not replying to this issue in time.
I also tried to launch the developer console to check the logs when the exception occurred, but when I tried again it was successful and could not reproduce the situation.
Since the DNS is not stable where I was at the time of the problem, I suspect it was a network issue, maybe you can test it with a tool like fiddler or charles, I remember there are similar tools in the developer console to simulate bad network.
Or if there is a local log that can tell me the location, I can try to find and provide relevant files.

**Comment by sockejr** [comment]
> @sockejr @ZPOMI @franoisD thanks for reporting. If you open developer tools in Podman Desktop (CTRL + SHIFT + I), open console, there could be an error when executing the workflow, can you please past it here. Or alternatively, open troubleshooting page and check Logs (PD status bar light bulb icon -> Logs).

Output from Developer Tools: 
Can't get machine podman-machine-default resource usage error Error: no running provider for the matching container

![image](https://github.com/containers/podman-desktop/assets/50873185/aba3f241-0687-4d1c-bb78-7bac35d154a4)

**Comment by sockejr** [comment]
> > @sockejr @ZPOMI @franoisD thanks for reporting. If you open developer tools in Podman Desktop (CTRL + SHIFT + I), open console, there could be an error when executing the workflow, can you please past it here. Or alternatively, open troubleshooting page and check Logs (PD status bar light bulb icon -> Logs).
> 
> Thank you very much for your work. I'm sorry for not replying to this issue in time. I also tried to launch the developer console to check the logs when the exception occurred, but when I tried again it was successful and could not reproduce the situation. Since the DNS is not stable where I was at the time of the problem, I suspect it was a network issue, maybe you can test it with a tool like fiddler or charles, I remember there are similar tools in the developer console to simulate bad network. Or if there is a local log that can tell me the location, I can try to find and provide relevant files.

From the Troubleshooting page the following is in the logs: 
main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT //./pipe/docker_engine
main ↪️ Activating extension (podman-desktop.kube-context) ended in 1077 milliseconds
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 731.6700000017881ms
main ↪️ [kind] kind extension is active
main ↪️ Activating extension (podman-desktop.kind) ended in 1339 milliseconds
main ↪️ [kind] kind extension is active
main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT //./pipe/docker_engine
main ↪️ Activating extension (podman-desktop.podman) ended in 3418 milliseconds
main ↪️ PluginSystem: initialization done.
main ↪️ Autostarting podman-desktop.podman container engine
main ↪️ [podman] Podman extension: Autostarting machine podman-machine-default
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 159.3378000035882ms
main ↪️ [kind] kind extension is active
main ↪️ [podman] Can't get machine podman-machine-default resource usage error Error: no running provider for the matching container
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 96.91070000082254ms
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 139.24430000782013ms
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 170.50519999861717ms
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 73.54919999837875ms
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 158.74860000610352ms
main ↪️ [compose] Successfully installed 'docker-compose' binary.
main ↪️ [compose] Successfully installed 'docker-compose' binary.

**Comment by sockejr** [comment]
I updated podman Desktop to 1.7.1 and Podman to 4.9.2 same issue, here the logs wehen trying to instal podman-compose:
log : main ↪️ Error while fetching API groups: Error: self signed certificate in certificate chain
log : main ↪️ Error while fetching API groups: Error: self signed certificate in certificate chain
warn : MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 extensions-started listeners added. Use emitter.setMaxListeners() to increase limit
warn : MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 extension-stopped listeners added. Use emitter.setMaxListeners() to increase limit
log : main ↪️ PluginSystem: received dom-ready event from the UI
log : main ↪️ Starting http server to handle webviews on port 44000
log : main ↪️ Delayed startup, flushing 11 events
log : main ↪️ System ready. Loading extensions...
log : main ↪️ Activating extension (podman-desktop.compose) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.docker) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.kind) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.kube-context) with max activation time of 10 seconds
log : main ↪️ [kube-context] starting extension kube-context
log : main ↪️ Activating extension (podman-desktop.kubectl-cli) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.lima) with max activation time of 10 seconds
debug : main ↪️ [lima] Could not find socket at C:\Users\user\.lima\podman\sock\podman.sock
log : main ↪️ Activating extension (podman-desktop.podman) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.registries) with max activation time of 10 seconds
log : main ↪️ Activating extension (podman-desktop.docker) ended in 93 milliseconds
log : main ↪️ Activating extension (podman-desktop.lima) ended in 38 milliseconds
log : main ↪️ Activating extension (podman-desktop.registries) ended in 8 milliseconds
log : main ↪️ Activating extension (podman-desktop.compose) ended in 136 milliseconds
log : main ↪️ Activating extension (podman-desktop.kubectl-cli) ended in 85 milliseconds
warn : main ↪️ [kubectl-cli] Error getting kubectl from user PATH: Error: Failed to execute command: spawn kubectl.exe ENOENT, trying from extension storage path
warn : main ↪️ [kubectl-cli] Error getting kubectl version system from extension storage path Error: Failed to execute command: spawn C:\Users\user\.local\share\containers\podman-desktop\extensions-storage\podman-desktop.kubectl-cli\bin\kubectl.exe ENOENT
debug : main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT //./pipe/docker_engine
log : main ↪️ Activating extension (podman-desktop.kube-context) ended in 483 milliseconds
log : main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 524.9976999759674ms
log : main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 760.0159999728203ms
log : main ↪️ [kind] kind extension is active
log : main ↪️ Activating extension (podman-desktop.kind) ended in 751 milliseconds
log : main ↪️ [kind] kind extension is active
debug : main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT //./pipe/docker_engine
log : main ↪️ Activating extension (podman-desktop.podman) ended in 3636 milliseconds
log : main ↪️ PluginSystem: initialization done.
log : main ↪️ Autostarting podman-desktop.podman container engine
log : main ↪️ [podman] Podman extension: Autostarting machine podman-machine-default
log : main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 116.90110003948212ms
warn : main ↪️ [podman]  Can't get machine podman-machine-default resource usage error Error: no running provider for the matching container
log : main ↪️ [kind] kind extension is active
error : main ↪️ [podman] Error during install!
error : main ↪️ [podman] Error: Command execution failed with exit code 3010
error : main ↪️ /event stream received an error. Error: aborted
warn : main ↪️ Error when handling events Error: Error in handling events Will reconnect in 5s Error: Error in handling events
log : main ↪️ error is Error: connect ENOENT \\.\pipe\podman-machine-default
warn : main ↪️ Error when handling events Error: Error in handling events Will reconnect in 5s Error: Error in handling events
debug : main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT //./pipe/docker_engine
log : main ↪️ Aborting reconnect due to error as connection is now stopped
log : main ↪️ Aborting reconnect due to error as connection is now stopped
log : main ↪️ Aborting reconnect due to error as connection is now stopped
warn : main ↪️ [podman]  Can't get machine podman-machine-default resource usage error Error: no running provider for the matching container
error : main ↪️ /event stream received an error. Error: aborted
warn : main ↪️ Error when handling events Error: Error in handling events Will reconnect in 5s Error: Error in handling events
error : main ↪️ /event stream received an error. Error: aborted
warn : main ↪️ Error when handling events Error: Error in handling events Will reconnect in 5s Error: Error in handling events
warn : main ↪️ [podman]  Can't get machine podman-machine-default resource usage error Error: no running provider for the matching container
log : main ↪️ error is Error: connect ENOENT \\.\pipe\podman-machine-default
warn : main ↪️ Error when handling events Error: Error in handling events Will reconnect in 5s Error: Error in handling events
log : main ↪️ error is Error: connect ENOENT \\.\pipe\podman-machine-default
warn : main ↪️ Error when handling events Error: Error in handling events Will reconnect in 5s Error: Error in handling events
debug : main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT //./pipe/docker_engine
log : main ↪️ Aborting reconnect due to error as connection is now stopped
log : main ↪️ Aborting reconnect due to error as connection is now stopped
warn : main ↪️ [podman]  Can't get machine podman-machine-default resource usage error Error: no engine matching this container
log : main ↪️ [compose] Successfully installed 'docker-compose' binary.

**Comment by castaglio** [comment]
I was getting the same error when setting up compose.  "Failed installing Compose".

In spite of that I tried to run `podman compose` to not avail.  Getting error 
```
2 errors occurred:
        * exec: "docker-compose": executable file not found in %PATH%
        * exec: "podman-compose": executable file not found in %PATH%
 ```
You can see in the image that it says "Successfully installed 'docker-compose' binary" though.

![PodmanVersionAndError](https://github.com/containers/podman-desktop/assets/7218843/2fc98b7b-b21d-46ba-9f90-cb2cb5d4ff70)

**Workaround :**
From another machine where "podman compose" is working I've got the binary path.
%USERPROFILE%\AppData\Local\Microsoft\WindowsApps

Then I confirmed that it was indeed installed.

Added the path to %PATH% and restarted the terminal and now is working at least in the terminal although it still shows pending to setup in "Podman Desktop".

**Comment by Axeltherabbit** [comment]
I have the same issue with the following error:

```
Onboarding.svelte:222 Error: Error invoking remote method 'command-registry:executeCommand': Error: An object could not be cloned.
N @ Onboarding.svelte:222
y @ Onboarding.svelte:209
await in y (async)
_ @ Onboarding.svelte:165
O @ Onboarding.svelte:268
await in O (async)
A @ Onboarding.svelte:257
le @ Onboarding.svelte:497
(anonymous) @ lifecycle.js:181
Gi @ lifecycle.js:181
b @ Button.svelte:34
Show 2 more frames
Show less
VM5:2 main ↪️ Error occurred in handler for 'command-registry:executeCommand': Error: An object could not be cloned.
    at WebContents.<anonymous> (node:electron/js2c/browser_init:2:82839)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
console.<computed> @ VM5:2
(anonymous) @ VM5:2
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2

```

**Comment by arthurcaccavo** [comment]
having the same error, when trying to install I get the following messages
![Image](https://github.com/user-attachments/assets/fcc6fbab-4d68-4ab0-beca-ee9e8fb744c7)

**Comment by carlosonunez** [comment]
Hi! I ran into this problem today on a Windows 11 24H2 machine running within Azure. The error from Dev Tools states this:

```
Error: No asset found for windows and x86_64
```

However, Docker does maintain an `windows_x86_64` variant of the Compose binary here: https://github.com/docker/compose/releases/download/v2.33.1/docker-compose-windows-x86_64.exe.

Furthermore, downloading this artifact manually and moving it to `$env:USERPROFILE\.local\share\containers\podman-desktop\extensions-storage\podman-desktop.compose\bin\docker-compose.exe` works around this issue.

From looking at the release asset downloading code [here](https://github.com/podman-desktop/podman-desktop/blob/a4c9e16183cb2b60ca60a793e05e4d36a41b6ead/extensions/compose/src/compose-github-releases.ts#L87), I suspected that this is due to GitHub HTTP 403 or HTTP 429'ing API requests from cloud VMs. However, manually executing the `/releases` API request with `Invoke-WebRequest` yields a 200 OK with a valid JSON payload, so I'm confused as to how this error happens.
